analyzer/src/main/java/nl/tudelft/Runner.java

```diff
@@ -47,126 +27,50 @@ Runner addExtractor(Extractor extractor) throws SQLException {
         return this;
     }
 
-    void clear(PackageId[] packages) {}
-
-    void run(Maven mvn, Collection<? extends PackageId> packages, Map<PackageId, String> packagingTypes, Config config) {
+    void run(Maven mvn, Collection<? extends ArtifactId> packages, int threads) throws SQLException, InterruptedException {
         var fields = extractors.values().stream()
                 .map(Extractor::fields)
                 .flatMap(Arrays::stream)
                 .toArray(Field[]::new);
         if (fields.length == 0)
             return;
-        processPackages(packages, fields, mvn, packagingTypes, config);
-    }
-
-    private void processPackages(Collection<? extends PackageId> packages, Field[] fields, Maven mvn, Map<PackageId, String> packagingTypes, Config config) {
-        ExecutorService executor = Executors.newFixedThreadPool(config.getThreads());
-
-        // We manually create then manage the future inside the task
-        // since executor.submit() only returns a Future, but we need
-        // a CompletableFuture to be able to use CompletableFutures.allOf()
-        List<Future<Void>> futures = new ArrayList<>();
-        for (PackageId id : packages) {
-            String pkgType = packagingTypes.get(id);
-            CompletableFuture<Void> future = new CompletableFuture<>();
-            executor.submit(new ProcessPackageTask(id, fields, mvn, future, pkgType));
-            futures.add(future);
-        }
 
-        // Combine all CompletableFutures into a single CompletableFuture then .get() to wait for all threads
-        // to complete (successfully or exceptionally)
+        db.createUnresolvedTable(false);
+        var pool = new ForkJoinPool(threads);
         try {
-            CompletableFuture.allOf(futures.toArray(new CompletableFuture[0])).get();
-        } catch (InterruptedException e) {
-            Thread.currentThread().interrupt();
-            LOGGER.error(e);
+            pool.submit(() -> packages.parallelStream().forEach(id -> execute(mvn, fields, id))).get();
         } catch (ExecutionException e) {
-            // Whenever a package exception occurs at any point during execution,
-            // it will be logged here, so ignore it.
-            LOGGER.error(e);
-        }
-        finally {
-            executor.shutdown();
+            throw new RuntimeException(e);
```

Comment:
ExecutionException already wraps an exception. Consider changing this to `throw new RuntimeException(e.getCause())`.